crates/oxc_linter/src/rules/jsx_a11y/aria_props.rs

```diff
@@ -0,0 +1,156 @@
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use std::collections::HashSet;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jsx-a11y(aria-props):")]
+#[diagnostic(severity(warning), help("{1}"))]
+struct AriaPropsDiagnostic(#[label] pub Span, String);
+
+#[derive(Debug, Default, Clone)]
+pub struct AriaProps;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that elements do not use invalid ARIA attributes.
+    ///
+    /// ### Why is this bad?
+    /// Using invalid ARIA attributes can mislead screen readers and other assistive technologies.
+    /// It may cause the accessibility features of the website to fail, making it difficult
+    /// for users with disabilities to use the site effectively.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <input aria-labeledby="address_label" />
+    ///
+    /// // Good
+    /// <input aria-labelledby="address_label" />
+    /// ```
+    AriaProps,
+    correctness
+);
+
+impl Rule for AriaProps {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let valid_aria_props = create_valid_aria_props();
+
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            for attr in &jsx_el.attributes {
+                if let JSXAttributeItem::Attribute(attr) = attr {
+                    let name = get_attribute_name(&attr.name).to_lowercase();
+                    if name.starts_with("aria-") && !valid_aria_props.contains(&name) {
+                        let error_message =
+                            format!("{}: This attribute is an invalid ARIA attribute.", name);
+                        ctx.diagnostic(AriaPropsDiagnostic(attr.span, error_message));
+                    }
+                }
+            }
+        }
```

Comment:
i think we can probably do

```rs
if let AstKind::JSXAttributeItem(attr) = node.kind() {
```

this will avoid a duplicate loop as currently, the linter will loop over over all of those properties as well as your code

Reply:
This is indeed a good optimization. Thank you! https://github.com/oxc-project/oxc/pull/1797/commits/3de2f5b1dbf8d7e40617355d267ef0c5dbafb1ce